### PR TITLE
Feature/d 10 readiness

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/HRIN-754

Preparing for d 10.

- Make custom modules and themes pass code analysis and compatible with d10
- Update contrib modules (Major versions):
    - better_exposed_filters
    - captcha
    - color_field
    - openid_connect
    - search_autocomplete
    - toolbar_visibility
    - twig_tweak
    - viewsreference

### CODE REVIEW

File: web/modules/custom/hoeringsportal_data/src/Controller/Api/HearingController.php [hunk 1]
```diff
@@ -38,7 +38,7 @@ public function show($hearing) {
 
     $entity = \reset($entities);
 
-    $data = $this->serialize($entity);
+    $data = $this->serializer->serialize($entity, 'json');
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_data/src/Controller/Api/HearingController.php:41]: 🎉

**Review by rimi-itk — CHANGES_REQUESTED** [review]
Looks good. 

We should go for just `core_version_requirement: ^10` in custom modules to keep things clean.

File: web/modules/custom/aarhus_hero/aarhus_hero.info.yml [hunk 2]
```diff
@@ -4,4 +4,4 @@ interface translation project: aarhus_hero
 interface translation server pattern: modules/custom/aarhus_hero/translations/aarhus_hero.%language.po
 description: 'Adds a config page and a block for the hero.'
 package: Aarhus modules
-core_version_requirement: ^8.8 || ^9
+core_version_requirement: ^8.8 || ^9 || ^10
```
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/aarhus_hero/aarhus_hero.info.yml:7]: I'd prefer to make the site completely ready on d9 before making the actual update to 10. The 2nd step should include the actual ^9 -> ^10 and deleting modules that was disabled in step one. I guess we could hold the info.yml changes  (^10 only) to that step as well. But for this code to work now on d9 we need it.

File: web/modules/custom/hoeringsportal_project_timeline/assets/public/build/manifest.json
[the remark this answers is not in the document]
Comment (martinyde) [reply web/modules/custom/hoeringsportal_project_timeline/assets/public/build/manifest.json:1]: I think that is for a separate issue.

File: web/modules/custom/hoeringsportal_base_fixtures/src/Helper/Helper.php [hunk 3]
```diff
@@ -79,7 +79,12 @@ public function createDocumentsFromAssets(): array {
     $this->fileSystem->prepareDirectory($document_target_path, FileSystemInterface:: CREATE_DIRECTORY | FileSystemInterface::MODIFY_PERMISSIONS);
 
     // Loop over documents to add them properly to the file system.
-    foreach (glob($document_source_path . '/*.{pdf,docx}', GLOB_BRACE) as $document) {
+    foreach (glob($document_source_path . '/*.pdf') as $document) {
+      $destination = $this->fileSystem->copy($document, $document_target_path . '/' . basename($document), FileSystemInterface::EXISTS_REPLACE);
+      $documents[] = $destination;
+    }
+
+    foreach (glob($document_source_path . '/*.docx') as $document) {
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_base_fixtures/src/Helper/Helper.php:87]: This code duplicates the PDF stuff above. Consider rewriting to 

```php
$extensions = ['pdf', 'docx'];
foreach ($extensions as $extension) {
  foreach (glob($document_source_path . '/*.' . $extension) as $document) {
    $destination = $this->fileSystem->copy($document, $document_target_path . '/' . basename($document), FileSystemInterface::EXISTS_REPLACE);
    $documents[] = $destination;
  }
}
```

**Review by rimi-itk — APPROVED** [review]
Approved with a comment.

**Pull request merged**